src/compiler/compiler.rs

```diff
@@ -1283,14 +1318,65 @@ LLVM version: 6.0",
         );
         // rustc --print=sysroot
         let sysroot = f.tempdir.path().to_str().unwrap();
-        next_command(&creator, Ok(MockChild::new(exit_status(0), &sysroot, "")));
-        next_command(&creator, Ok(MockChild::new(exit_status(0), &sysroot, "")));
-        next_command(&creator, Ok(MockChild::new(exit_status(0), &sysroot, "")));
-        let c = detect_compiler(creator, &rustc, f.tempdir.path(), &[], pool, None)
-            .wait()
-            .unwrap()
-            .0;
+        next_command(creator, Ok(MockChild::new(exit_status(0), &sysroot, "")));
+        next_command(creator, Ok(MockChild::new(exit_status(0), &sysroot, "")));
+        next_command(creator, Ok(MockChild::new(exit_status(0), &sysroot, "")));
+    }
+
+    #[test]
+    fn test_detect_compiler_kind_rustc_workspace_wrapper() {
+        let f = TestFixture::new();
+        // Windows uses bin, everything else uses lib. Just create both.
+        fs::create_dir(f.tempdir.path().join("lib")).unwrap();
+        fs::create_dir(f.tempdir.path().join("bin")).unwrap();
+        let rustc = f.mk_bin("rustc-workspace-wrapper").unwrap();
+        let creator = new_creator();
+        let runtime = single_threaded_runtime();
+        let pool = runtime.handle();
+        populate_rustc_command_mock(&creator, &f);
+        let c = detect_compiler(
+            creator,
+            &rustc,
+            f.tempdir.path(),
+            &[OsString::from("rustc")],
```

Comment:
If the above could be different from `rustc`, we should verify that as well.

Reply:
I've added a function for detecting rustc, and added a tests for that.